SSA/Projects/FullyHomomorphicEncryption/Basic.lean

```diff
@@ -81,72 +142,240 @@ abbrev R := (ZMod q)[X] ⧸ (Ideal.span {f q n})
 -- Coefficients of `a : R' q n` are `a\_i : Zmod q`.
 -- TODO: get this from mathlib
 
-/-- Canonical epimorphism `ZMod q[X] ->*+ R q n` -/
+/-- Canonical epimorphism / quotient map: `ZMod q[X] ->*+ R q n` -/
 abbrev R.fromPoly {q n : Nat} : (ZMod q)[X] →+* R q n := Ideal.Quotient.mk (Ideal.span {f q n})
```

Comment:
There's an elaborator improvement in the latest Lean that might make this faster if you don't have that version already.

Reply:
This definition, or lemmas about it. There always seemed to be problems around quotients.